tasks/sign-binaries/tests/mac_mocks.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+
+count_file=$(workspaces.data.path)/mac_ssh_count.txt
+if [[ ! -f "$count_file" ]]; then
+    echo "0" > "$count_file"
+fi
+
+
+function ssh() {
+    # Read the current ssh_call_count from the file
+    ssh_call_count=$(cat "$count_file")
+    ssh_call_count=$((ssh_call_count + 1))
+    echo "$ssh_call_count" > "$count_file"
+
+    echo "$ssh_call_count" > $(workspaces.data.path)/ssh_calls_mac.txt
+}
+
+scp_count_file="/tmp/scp_count_mac.txt"
+if [[ ! -f "$scp_count_file" ]]; then
+    echo "0" > "$scp_count_file"
+fi
+
+function scp() {
+    scp_call_count=$(cat "$scp_count_file")
+    scp_call_count=$((scp_call_count + 1))
+    echo "$scp_call_count" > "$scp_count_file"
+    if [[ "$scp_call_count" -eq 1 ]]; then
+        echo "$@" > "$(workspaces.data.path)/mock_scp_1.txt"
+    fi
+
+    scp_args=$@
+
+    # Check for Mac signing script
+    if [[ "$scp_args" == *"mac_signing_script"* ]]; then
+        echo "MAC SCP command detected with args: $@"
+        echo "$@" > $(workspaces.data.path)/scp_mac_args.txt
+    fi
+
+    # Check for Windows signing script
```

Comment:
I'm not sure exactly how the various mock files differ, but I can see that this one is only used for the mac signing step, so this shouldn't be here, right? And then `mocks.sh` is not used for mac signing, so it shouldn't have the mac signing script condition.

Reply:
Main reason is to keep them separate, since we were working on separate step implementation that would cause less conflicts.
You are correct about extra windows and mac tests, I updated that in the latest commit, ty!